### PR TITLE
Refactores publish, clone, delete actions from product grid - for DRY principle

### PR DESCRIPTION
**Pull request by ramusus**
fixed issues with Alerts during operations with group of selected products, cleaned the code for the sake of DRY principle.

### CODE REVIEW

**Comment by zenweasel** [comment]
@ramusus Some ESLint issues with this as well. If you wouldn't mind fixing this up. (You can usually take the identical code notes as warnings if they are legitimate)

File: packages/reaction-core/client/helpers/globals.js [hunk 1]
```diff
@@ -72,48 +72,104 @@ this.getProductsByTag = function (tag) {
...
-  let confirmTitle = "Delete this product?";
-
-  if (_.isArray(product)) {
-    if (product.length === 1) {
-      title = product[0].title || "the product";
-      productIds = [product[0]._id];
-    } else {
-      title = "the selected products";
-      confirmTitle = "Delete selected products?";
+this.publishProduct = function (productOrArray) {
```
Comment (aaronjudd) [inline packages/reaction-core/client/helpers/globals.js]: should this just be `publishProduct` and not `this.publishProduct` where `this.publishProduct` is window scope, and `publishProduct` should be package scoped.

File: packages/reaction-core/client/helpers/globals.js [hunk 2]
```diff
@@ -72,48 +72,104 @@ this.getProductsByTag = function (tag) {
...
+  });
+};
+
+/**
+ * maybeDeleteProduct
+ * @summary confirm product deletion, delete, and alert
+ * @param {Object} productOrArray - product Object
+ * @return {Object} - returns nothing, and alerts, happen here
+ */
+this.maybeDeleteProduct = function (productOrArray) {
```
Comment (aaronjudd) [inline packages/reaction-core/client/helpers/globals.js]: `maybeDeleteProduct = function (productOrArray) {`

File: packages/reaction-core/client/helpers/globals.js [hunk 3]
```diff
@@ -72,48 +72,104 @@ this.getProductsByTag = function (tag) {
...
-      productIds = _.map(product, (item) => {
-        return item._id;
+/**
+ * cloneProduct
+ * @summary product cloning and alert
+ * @param {Object} productOrArray - product Object
+ * @param {Boolean} redirect - shoud we do redirect after cloning?
+ * @returns {any}
+ */
+this.cloneProduct = function(productOrArray, redirect) {
```
Comment (aaronjudd) [inline packages/reaction-core/client/helpers/globals.js]: `cloneProduct = function(productOrArray, redirect) {`

Comment (ramusus) [inline packages/reaction-core/client/helpers/globals.js]: fixed

File: packages/reaction-core/client/helpers/globals.js (hunk 2 above)
Comment (ramusus) [inline packages/reaction-core/client/helpers/globals.js]: fixed

File: packages/reaction-core/client/helpers/globals.js (hunk 1 above)
Comment (ramusus) [inline packages/reaction-core/client/helpers/globals.js]: fixed

**Comment by ramusus** [comment]
@zenweasel I cleaned the code up.

File: packages/reaction-core/client/helpers/globals.js [hunk 4]
```diff
@@ -72,48 +72,104 @@ this.getProductsByTag = function (tag) {
...
+      Alerts.add(i18n.t("productDetail.clonedAlert") + " " + product.title, "success", {
+        placement: "productGridItem",
+        id: product._id,
+        autoHide: true,
+        dismissable: false
+      });
+    }
+    if (redirect) {
+      Router.go("product", {
+        _id: productId
```
Comment (johannes-scharlach) [inline packages/reaction-core/client/helpers/globals.js]: After this refactoring, `productId` doesn't seem to be defined anymore. (I'm new to this codebase, forgive me if I'm wrong)

Comment (johannes-scharlach) [inline packages/reaction-core/client/helpers/globals.js]: As far as I see it is implied if there should be a redirect: It should only happen, if there was a single product passed to this function, not if it's an array. So I would suggest the following:

```js
/**
 * cloneProduct
 * @summary product cloning and alert
 * @param {Object} productOrArray - product Object
 * @returns {Object} - returns nothing, and alerts, happen here
 */
cloneProduct = function (productOrArray) {
  const paramIsProduct = ! _.isArray(productOrArray);
  const products = paramIsProduct ? [productOrArray] : productOrArray;

  return Meteor.call("products/cloneProduct", products, function (error) {
    if (error) {
      throw new Meteor.Error("error cloning product", error);
    }
    for (let product of products) {
      Alerts.add(i18n.t("productDetail.clonedAlert") + " " + product.title, "success", {
        placement: "productGridItem",
        id: product._id,
        autoHide: true,
        dismissable: false
      });
    }

    if (paramIsProduct) {
      Router.go("product", {
        _id: productOrArray._id
      });
    }
  });
};
```

Comment (ramusus) [inline packages/reaction-core/client/helpers/globals.js]: Yes, thanks for suggestion! I refactored a little bit more

**Comment by zenweasel** [comment]
@ramusus Hey Boris, looks like there are still a couple of linting errors left, plus there are conflicts that need to be resolved before we can merge.

**Comment by zenweasel** [comment]
Ok to merge pending conflict resolution and fixing the "else" statement, rest of the linting errors are ok.

**Comment by ramusus** [comment]
@zenweasel I fixed "else" statement and removed few "identical code" issues. What do you mean about "pending conflict resolution"?

**Comment by zenweasel** [comment]
@ramusus This PR has conflicts with the current development branch. (You can see that  below where it says "This branch has conflicts that must be resolved"). You just need to merge `development` back into your branch and resolve any conflicts and then push that back and we should be ready to merge this in.

If you can try to do this quickly that would be great, because once @aaronjudd starts merging in v12, all hell will break loose and I can't guarantee who will make it out alive. ;)

**Comment by ramusus** [comment]
Yes, understood. Thanks for explanation. Merged with development. 
Yes, v12 already gives me a lot of work for fixing integration with my packages.

**Pull request merged**